internal/controller/components/kserve/kserve.go

```diff
@@ -43,6 +43,7 @@ var (
 	conditionTypes = []string{
 		status.ConditionServingAvailable,
 		status.ConditionDeploymentsAvailable,
+		status.ConditionKserveConfigurationOptimal,
```

Comment:
I don't think this must be defined here, as then this condition is taken into account to compute the readiness of the component, whereas this is only a warning/info for what I get from the PR

Reply:
I agree with @lburgazzoli's concern here. Adding this condition to the `conditionTypes` array will contribute to the component's readiness calculation, but it should be a warning/informational condition only.